File: Leetcode-Problems/Lemonade Change.cpp
class Solution
{
public:
    bool lemonadeChange(vector<int> &bills)
    {
        int five = 0, ten = 0;
        for (int i = 0; i < bills.size(); i++)
        {
            if (bills[i] == 5)
            {
                five++;
            }
            else if (bills[i] == 10)
            {
                if (five == 0)
                    return false;
                else
                {
                    ten++;
                    five--;
                }
            }
            else if (bills[i] == 20)
            {
                if (ten == 0)
                {
                    if (five >= 3)
                    {
                        five -= 3;
                    }
                    else
                        return false;
                }
                else if (ten == 0 && five == 0)
                    return false;
                else if (ten != 0)
                {
                    if (five >= 1)
                    {
                        ten--;
                        five--;
                    }
                    else
                    {
                        return false;
                    }
                }
            }
        }
        return true;
    }
};

File: Leetcode-Problems/Lemonade Change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int fives = 0, tens = 0;
        for (int i = 0; i < bills.length; i++) {
            if (bills[i] == 5)
                fives++;
            else if (bills[i] == 10) {
                if (fives > 0) {
                    fives--;
                } else {
                    return false;
                }
                tens++;
            } else {
                if (tens > 0) {
                    if (fives > 0) {
                        tens--;
                        fives--;
                    } else {
                        return false;
                    }
                } else {
                    if (fives > 2) {
                        fives -= 3;
                    } else
                        return false;
                }
            }
        }
        return true;
    }
}